dialog/file.go

```diff
@@ -206,7 +206,13 @@ func (f *fileDialog) makeUI() fyne.CanvasObject {
 		title = label + " Folder"
 	}
 
-	f.setView(gridView)
+	viewPreference := fyne.CurrentApp().Preferences().StringWithFallback("fileDialogViewLayout", "grid")
+	if viewPreference == "grid" {
+		f.setView(gridView)
+	} else {
+		f.setView(listView)
+	}
```

Comment:
I don't think we gain much from using a string for this. Sure, it makes the JSON a bit more human readable but I don't think it is very useful? It is an internal detail after all. If you use an int for the preference, you can avoid the if-statement when loading it into the setView. It will also default to 0 by default so no need to load with a default.
```suggestion
	viewPreference := fyne.CurrentApp().Preferences().Int("fileDialogViewLayout")
	f.setView(viewLayout(viewPreference))
```

Reply:
I was back and forth on this but I think int does make sense since it is internal. I have updated to make this change.